src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -56,7 +56,7 @@ public partial class Process : Component
         private bool _priorityBoostEnabled;
         private bool _havePriorityBoostEnabled;
 
-        private bool _raisedOnExited;
+        private volatile bool _raisedOnExited;
```

Comment:
This one, especially, would possibly be changeable to some form of `Interlocked.CompareExchange`... except the code is using `lock(this)` (arrghh), which is at least a partial breaking change, and might have wider effects as well.